File: python3/RemoveElementsInLinkedListRecursive.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        if head == None:
            return head
        h = self.removeElements(head.next, val)
        
        if head.val == val:
            return h
        else:
            head.next = h
        return head


 

File: java/RemoveElementsInLinkedListRecursive.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
     public ListNode removeElements(ListNode head, int val) {
        if(head == null) return head;
        ListNode h = removeElements(head.next, val);
        
        if(head.val == val) return h;
        else head.next = h;
        
        return head;
    }
}
